packages/utilities/exposure-scanning/patch/index.js

```diff
@@ -0,0 +1,87 @@
+/* eslint-disable global-require */
+let octokit;
+
+(async () => {
+  if (process.env.GITHUB_ACTION && process.env.GITHUB_TOKEN) {
+    const { Octokit } = await import('@octokit/action');
+    octokit = new Octokit();
+  } else {
+    const { Octokit } = await import('@octokit/rest');
+    octokit = new Octokit();
+  }
+})();
+
+const patchPrBody = async (reqBody, body) => {
+  await octokit.request('PATCH /repos/{owner}/{repo}/pulls/{prId}', {
+    ...reqBody,
+    body,
+  });
+};
+
+const patchPrReviewComments = async (reqBody, comments) => {
+  await Promise.all(
+    comments.map(async ({ id, body }) => {
+      await octokit.request(
+        'PATCH /repos/{owner}/{repo}/pulls/comments/{commentId}',
+        {
+          owner: reqBody.owner,
+          repo: reqBody.repo,
+          commentId: id,
+          body,
+        },
+      );
+    }),
+  );
+};
```

Comment:
I don't think you're supposed to await for a promise to resolve in each iteration like this. This would mean that you need to wait for the request to finish before the next iteration happens and so on until all items have been mapped. This is quite slow compared to correctly using a Promise.all where all requests happen at once e.g.

```suggestion
const patchPrReviewComments = async (reqBody, comments) =>
  Promise.all(
    comments.map(({ id, body }) =>
      octokit.request(
        'PATCH /repos/{owner}/{repo}/pulls/comments/{commentId}',
        {
          owner: reqBody.owner,
          repo: reqBody.repo,
          commentId: id,
          body,
        },
      ),
    ),
  );
```

Reply:
Ahhh yeah that makes sense! My JS async knowledge is very limited 😂 Thanks 🙌